prov/efa/src/rxr/rxr_rma.c

```diff
@@ -631,6 +631,27 @@ ssize_t rxr_rma_inject_write(struct fid_ep *ep, const void *buf, size_t len,
 	struct fi_msg_rma msg;
 	struct iovec iov;
 	struct fi_rma_iov rma_iov;
+	struct rxr_ep *rxr_ep;
+	struct rdm_peer *peer;
+
+	rxr_ep = container_of(ep, struct rxr_ep, util_ep.ep_fid.fid);
+	peer = rxr_ep_get_peer(rxr_ep, dest_addr);
+	if (peer->is_local) {
+		/* When fi_inject_write() is used, libfabric does not have to write a completion.
+		 * Hence it is more efficient to use shm provider's fi_inject_write() directly
+		 * than calling rxr_rma_writemsg().
+		 *
+		 * This is because rxr_rma_writemsg() will call shm's fi_writemsg(), for which shm
+		 * provider will generate a completion. There are two performance issues with that:
+		 *
+		 * First, to process the completion from shm, EFA needs to assigned a context to
+		 * fi_writemsg(). For that, EFA will allocate a tx_entry and and a pkt_entry, and
+		 * use the pkt_entry as context.
+		 *
+		 * Second, shm provider will use less efficient protocol when it needs to write a completion.
+		 */
```

Comment:
Is this comment really necessary? IMHO it's sufficient to have the info in the commit message.

Reply:
Same feeling. Or just keep the 1st paragraph is sufficient.

Reply:
The comment is for someone that does not read commit message.  Will remove the 2nd appearance and refer  the 1st one.

Reply:
It is straightforward to call fi_inject_write() to implement inject write, why would someone raise question here?

Reply:
I understand that you want to document the reason behind the change, but that is well served by the commit message. The code itself is clear enough to show the intention of special case optimization.